app/models/bookmark.rb

```diff
@@ -1,4 +1,13 @@
 class Bookmark < ApplicationRecord
-  belongs_to :user
-  belongs_to :tweet
+  belongs_to :user, inverse_of: :bookmarks
+  belongs_to :tweet, inverse_of: :bookmarks
+
+  validates :user_id, presence: true
+  validates :tweet_id, presence: true
+
+  # Association Validations
+  validates_associated :user, :tweet
+
+  # Scope to retrieve bookmarks made by a user
+  scope :user_bookmarks, ->(user_id) { where(user_id: user_id) }
```

Comment:
You have the same bookmark scope in the `User` model and in the `Tweet` model delete those scopes from there since it can lead to confusions